File: Algorithms/967.Numbers-With-Same-Consecutive-Differences/numbers_with_same_consecutive_differences.go
func numsSameConsecDiff(N int, K int) []int {
    num := []int{0, 1,2,3,4,5,6,7,8,9}
    for i:=0; i< N-1; i++ {
        tmp := []int{}
        for _, v := range num {
            if v == 0 && i == 0{
                continue
            }
            reminder := v % 10
            if reminder + K < 10 {
                tmp = append(tmp, v*10 + reminder + K)
            }
            if K > 0 && reminder - K >= 0 {
                tmp = append(tmp, v*10 + reminder - K)
            }
        }
        num = tmp
    }
    return num
}

File: Algorithms/967.Numbers-With-Same-Consecutive-Differences/numbers_with_same_consecutive_differences.py
class Solution:
    def numsSameConsecDiff(self, N: int, K: int) -> List[int]:
        if N == 1:
            return list(range(10))
        k_list = []
        for i in range(1, 10):
            if i + K < 10:
                k_list.append(i)
            if i - K >= 0:
                k_list.append(i)
    
        for i in range(1, N):
            tmp_res = list()
            for num in k_list:
                pre_ele = num % 10
                if pre_ele + K < 10:
                    tmp_res.append(num * 10 + pre_ele + K)
                if pre_ele - K >= 0:
                    if not (pre_ele - K == 0 and num == 0):
                        tmp_res.append(num * 10 + pre_ele - K)
            k_list = tmp_res         
        return list(set(k_list))   
    
